mono/profiler/log.c

```diff
@@ -3301,7 +3313,8 @@ handle_writer_queue_entry (void)
 		if (!entry->methods)
 			goto no_methods;
 
-		gboolean wrote_methods = FALSE;
+		gboolean wrote_methods;
+		wrote_methods = FALSE;
```

Comment:
These initialization changes make the code rather ugly. Why are they necessary?

Reply:
Cannot goto around initialization in C++.

Reply:
https://github.com/mono/mono/issues/10319#issuecomment-416873012